### PR TITLE
dockerized: handle podman auth files

### LINKED ISSUE #6260: `hack/dockerized`: podman auth file not visible from build container
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What happened**:
After doing a `podman login` and setting `DOCKER_PREFIX` to the corresponding account, `make push` is expected to work fine.
It's the case with docker thanks to:
https://github.com/kubevirt/kubevirt/blob/0294004fda493df8d2b8856e644915622cf6ace1/hack/dockerized#L110
By default, the location for the podman auth file looks like: `/tmp/podman-run-1001/containers/auth.json`, 1001 being the user ID.

**What you expected to happen**:
`make push` to use auth from `podman login`

**How to reproduce it (as minimally and precisely as possible)**:
$ podman login quay.io
$ export DOCKER_PREFIX="quay.io/mylogin"
$ make push

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
This should allow running `make push` when using podman as a CRI

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6260

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/dockerized [hunk 1]
```diff
@@ -107,9 +107,15 @@ _rsync \
...
-volumes="$volumes -v ${HOME}/.docker:/root/.docker:ro,z"
+# Copy latest docker/podman auth file and append .docker directory as volume
+mkdir -p ".docker"
+ID=$(id -u)
+if [[ $KUBEVIRT_CRI = "podman" ]] && [[ -f "/tmp/podman-run-${ID}/containers/auth.json" ]]; then
+    cp "/tmp/podman-run-${ID}/containers/auth.json" .docker/config.json
+elif [[ -f "${HOME}/.docker/config.json" ]]; then
+    cp "${HOME}/.docker/config.json" .docker/config.json
+fi
+volumes="$volumes -v $(pwd)/.docker:/root/.docker:ro,z"
```
Comment (davidvossel) [inline hack/dockerized:118]: i feel a bit weird about copying auth files around automatically into different directories.

What woudl be wrong with changing the host path on the volume to the podman authfile when podman is in use?

Comment (jean-edouard) [reply hack/dockerized:118]: Yeah I didn't like writing it either!
podman names the file differently (auth.json vs config.json), so I don't know how else to do this...

Comment (jean-edouard) [reply hack/dockerized:118]: Maybe we can do a single file volume or a bind mount to work around that, I'll give it a shot tomorrow.

Comment (jean-edouard) [reply hack/dockerized]: Force-pushed a bind mount, seems to do the trick!

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Pull request merged**